pkg/controller/utils/errorscrub.go

```diff
@@ -2,11 +2,15 @@ package utils
 
 import (
 	"regexp"
+	"strings"
 )
 
 var (
+	newlineTabRE = regexp.MustCompile(`(\n\t)`)
+	// aws
 	awsRequestIDRE = regexp.MustCompile(`(, )*(?i)(request id: )(?:[-[:xdigit:]]+)`)
-	newlineTabRE   = regexp.MustCompile(`(\n\t)`)
+	// azure
+	azureErrorDescriptionRE = regexp.MustCompile(`error_description\":\"([^\r\n]+)`)
```

Comment:
The `[^\r\n]+` part of this isn't doing what you expect, because a) those sequences are still escaped in the string (`\\r\\n`), and b) the square brackets represent alternation ("any single character that is neither a line feed nor a newline"). So, mostly by luck, `match[1]` is grabbing everything to the end of the string, and it's the `Split` on L27 that's cutting down to the part you care about.

I would suggest rewriting this as follows:
```suggestion
	azureErrorDescriptionRE := regexp.MustCompile(`\"error_description\":\"(.*?)\\r\\n`)
```
- First, I would match the leading `\"` in `\"error_description\"`. That's a nit.
- I'm matching `\\r\\n` (which is not the same as `\r\n`!) outside of the capture group...
- ...but using `?` to make the `.*` in the capture group non-greedy, so it only goes up to the first `\\r\\n`. (Normally we would do this with lookarounds, but go doesn't support them.)

Then below you don't need to do the `Split` -- see suggestion inline.

Reply:
This makes way more sense - updated. Thank you!!